tools/docs/config.ts

```diff
@@ -159,22 +213,30 @@ export async function generateConfig(dist: string, bot = false): Promise<void> {
     '\n'
   );
 
+  const indexed = indexMarkdown(configOptionsRaw);
+
   options
     .filter((option) => option.releaseStatus !== 'unpublished')
     .forEach((option) => {
       // TODO: fix types (#9610)
       const el: Record<string, any> = { ...option };
-      let headerIndex = configOptionsRaw.indexOf(`## ${option.name}`);
-      if (headerIndex === -1) {
-        headerIndex = configOptionsRaw.indexOf(`### ${option.name}`);
+
+      if (!indexed[option.name]) {
+        return;
```

Comment:
This is an error, docs should fail, so log an error and throw to abort

Reply:
this happens quite a lot, `options` from `lib/config/options/index.ts` contains config options from both `self-hosted-configuration.md` and `configuration-options.md` and we are doing one file at a time.

Reply:
Then we should check the `globalOnly` flag. Value should match `bot` variable.

Reply:
don't they have a header in one of the docs?

Reply:
so do but not in `self-hosted-configuration.md` and `configuration-options.md`. others have none, e.g. `bitbucket-pipelines`.

Reply:
This makes filtering by `globalOnly` a partial solution to this isnt it? ¯\\_(ツ)_/¯

Reply:
So our tests are wrong, every option should have a corresponding entry 🤔

Reply:
> So our tests are wrong, every option should have a corresponding entry 

i think we are good now, every option has an entry in `self-hosted-configuration.md` or `configuration-options.md`, otherwise its a manager that should have its own readme file which is covered by `lib/modules/manager/metadata.spec.ts`.